projects/pt1/python/torch_mlir_e2e_test/test_suite/norm_like.py

```diff
@@ -633,3 +633,73 @@ def forward(self, x, w, b):
 @register_test_case(module_factory=lambda: AtenInstanceNormModule())
 def AtenInstanceNormModule_basic(module, tu: TestUtils):
     module.forward(tu.rand(1, 2, 1, 3), tu.rand(2), tu.rand(2))
+
+
+# ==============================================================================
+class RenormModuleFloat32(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.p = 2
+        self.dim = 1
+        self.maxnorm = 10
+
+    @export
+    @annotate_args(
+        [
+            None,
+            ([3, 3], torch.float32, True),
+        ]
+    )
+    def forward(self, x):
+        return torch.ops.aten.renorm(x, self.p, self.dim, self.maxnorm)
+
+
+@register_test_case(module_factory=lambda: RenormModuleFloat32())
+def RenormModuleFloat32_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 3))
+
+
+class RenormModuleFloat16(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.p = 2.1
+        self.dim = 1
+        self.maxnorm = 10
+
+    @export
+    @annotate_args(
+        [
+            None,
+            ([3, 4, 5], torch.float16, True),
+        ]
+    )
+    def forward(self, x):
+        return torch.ops.aten.renorm(x, self.p, self.dim, self.maxnorm)
+
+
+@register_test_case(module_factory=lambda: RenormModuleFloat16())
+def RenormModuleFloat16_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 4, 5).to(torch.float16))
+
+
+class RenormModuleFloat32NegativeDim(torch.nn.Module):
```

Comment:
Does the conversion work with dynamic dims? I imagine so, but it would be nice if one of the tests could verify that.

Reply:
For this operator, dynamic dimensions do not work because the decomposition of AtenRenormOp involves the use of other operators, for example, AtenLinalgVectorNormOp. Ffor AtenLinalgVectorNormOp I need to calculate return Type (shape and dtype). To calculate the return shape, I need the input shape, which is just a list of -1 in case of dynamic dimensions. Additionally, at some point, I need to create a tensor of a certain size, and I calculate that size based on the input shape, which I don't have (just list of -1).
If is there some other solutions feel free to share your thoughts.

Reply:
@zjgarvey I added support for dynamic dims, can you take a look at the changes?